projects/notebooks/visualizations/rhods-notebooks-performance/analyze/__init__.py

```diff
@@ -1,13 +1,39 @@
-import matrix_benchmarking.common as common
+import json
 import logging
+import numpy as np
+
+from copy import deepcopy
+
+import matrix_benchmarking.common as common
+import matrix_benchmarking.regression as regression
 
 def run():
-    logging.info(f"Received {common.Matrix.count_records()} new entries")
-    for entry in common.Matrix.all_records(): pass
 
     logging.info(f"Received {common.LTS_Matrix.count_records()} historic LTS entries")
-    for lts_entry in common.LTS_Matrix.all_records(): pass
+    lts_entries_raw = list(common.LTS_Matrix.all_records())
+    # Temporarily skip the gathered results
+    lts_entries = list(filter(lambda entry: type(entry.results) is not list, lts_entries_raw))
+
+    logging.info(f"Received {common.Matrix.count_records()} new entries")
+    new_entries = list(common.Matrix.all_records())
 
     number_of_failures = 0
+    for entry in common.Matrix.all_records():
+        regression_results_dest = entry.location / "regression.json"
+
+        # Check for regression over the image
+        zscore = regression.ZScoreIndicator(
+            entry,
+            lts_entries,
+            settings_filter={"image": "tensorflow:2023.1"},
+            combine_funcs={"notebook_performance_benchmark_time": np.mean}
```

Comment:
can you use `notebook_performance_benchmark_min_time`, which already provides an "aggregated" value?
I think `min()` is more relevant than `mean()`, but that's open to discussion
if ever we chose to use `mean`, then we should expose it as a KPI instead of `_min`
(the idea is to expose as a KPI the value we're using for the regression analyses)

Reply:
I should have explained this better. The `combine_funcs={"notebook_performance_benchmark_time": np.mean}` is just an optional way of handling KPIs that provide either a `list[int]` or `list[float]`. The `combine_funcs` option can be omitted and any KPIs that use a list of values will be skipped. All the other KPIs such as `notebook_performance_benchmark_min_time` are analyzed as normal.